### PR TITLE
RDBC-282 Changes API doesn't reconnect after server failure

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -1106,7 +1106,16 @@ public async Task<ServerNode> HandleServerNotResponsive(string url, ServerNode c
...
             _nodeSelector?.OnFailedRequest(nodeIndex);
             var (_, serverNode) = await GetPreferredNode().ConfigureAwait(false);
-            await UpdateTopologyAsync(serverNode, 0, true, debugTag: "handle-server-not-responsive").ConfigureAwait(false);
+            try
+            {
+                await UpdateTopologyAsync(serverNode, 0, true, debugTag: "handle-server-not-responsive").ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                //We don't want to stop observe for changes if server down. we will wait for one to be up
```
Comment (ml054) [inline src/Raven.Client/Http/RequestExecutor.cs]: I think you mix classes responsibilities. You are in  HandleServerNotResponsive method in request executor, but comment is about changes api. So I think we should handle that in changes api instead.

File: src/Raven.Client/Documents/Changes/DatabaseChanges.cs [hunk 2]
```diff
@@ -483,8 +484,10 @@ private async Task DoWork()
                     // recover from the OnError accessing the faulty WebSocket.
                     try
                     {
-                        ConnectionStatusChanged?.Invoke(this, EventArgs.Empty);
+                        if (wasConnected)
+                            ConnectionStatusChanged?.Invoke(this, EventArgs.Empty);
 
+                        wasConnected = false;
                         _serverNode = await _requestExecutor.HandleServerNotResponsive(_url.AbsoluteUri, _serverNode, _nodeIndex, e).ConfigureAwait(false);
```
Comment (ml054) [inline src/Raven.Client/Documents/Changes/DatabaseChanges.cs]: I think we should catch here and handle the exception.

**Comment by ml054** [comment]
@efratshenhar Could you also write test case for that? It is quite important to avoid regressions in this piece of code.

**Comment by ml054** [comment]
by test case I mean:
1. STart single server
2. Connect to changes api
3. kill server
4. wait a bit
5. start server
6. modify document
7. wait for changes api to get notification

**Comment by efratshenhar** [comment]
yes

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
Merging, still need a test.

**Comment by talweiss1982** [comment]
https://github.com/ravendb/ravendb/pull/8799/

**Pull request merged**